src/redux/subtitleSlice.ts

```diff
@@ -252,8 +252,45 @@ export const selectHasChanges = (state: { subtitleState: { hasChanges: subtitle[
  * Will grab the state from videoState to skip past deleted segment if preview
  * mode is active.
  */
-export function setCurrentlyAtAndTriggerPreview(milliseconds: number) {
-  return (dispatch: AppDispatch, getState: () => RootState) => {
+// export function setCurrentlyAtAndTriggerPreview(milliseconds: number) {
+//   return (dispatch: AppDispatch, getState: () => RootState) => {
+//     milliseconds = roundToDecimalPlace(milliseconds, 0);
+
+//     if (milliseconds < 0) {
+//       milliseconds = 0;
+//     }
+
+//     const allStates = getState() as { videoState: video, subtitleState: subtitle; };
+//     const segments: Segment[] = allStates.videoState.segments;
+//     let triggered = false;
+
+//     if (allStates.subtitleState.isPlayPreview) {
+//       for (let i = 0; i < segments.length; i++) {
+//         if (segments[i].start < milliseconds && segments[i].end > milliseconds) {
+//           if (segments[i].deleted) {
+//             milliseconds = segments[i].end + 1;
+//             for (let j = i; j < segments.length; j++) {
+//               if (segments[j].deleted) {
+//                 milliseconds = segments[j].end + 1;
+//               } else {
+//                 break;
+//               }
+//             }
+//             triggered = true;
+//           }
+//           break;
+//         }
+//       }
+//     }
+
+//     dispatch(setCurrentlyAt(milliseconds));
+//     if (triggered) {
+//       dispatch(setPreviewTriggered(true));
+//     }
+//   };
+// }
```

Comment:
Is that a draft?

Reply:
Iirc I was testing out function definitions, this can go now.